src/matchers.rs

```diff
@@ -432,6 +433,63 @@ impl Match for BodyExactMatcher {
     }
 }
 
+#[derive(Debug)]
+/// Match part of the body of a request.
+///
+/// ### Example (string):
+/// ```rust
+/// use wiremock::{MockServer, Mock, ResponseTemplate};
+/// use wiremock::matchers::body_string_contains;
+///
+/// #[async_std::main]
+/// async fn main() {
+///     // Arrange
+///     let mock_server = MockServer::start().await;
+///
+///     Mock::given(body_string_contains("hello world"))
+///         .respond_with(ResponseTemplate::new(200))
+///         .mount(&mock_server)
+///         .await;
+///
+///     // Act
+///     let status = surf::post(&mock_server.uri())
+///         .body_string("this is a hello world example!".into())
+///         .await
+///         .unwrap()
+///         .status();
+///
+///     // Assert
+///     assert_eq!(status.as_u16(), 200);
+/// }
+/// ```
+pub struct BodyContainsMatcher(Vec<u8>);
+
+impl BodyContainsMatcher {
+    /// Specify the part of the body that should be matched as a string.
+    pub fn string<T: Into<String>>(body: T) -> Self {
+        Self(body.into().as_bytes().into())
+    }
+}
+
+/// Shorthand for [`BodyContainsMatcher::string`](struct.BodyContainsMatcher.html).
+pub fn body_string_contains<T>(body: T) -> BodyContainsMatcher
+where
+    T: Into<String>,
+{
+    BodyContainsMatcher::string(body)
+}
+
+impl Match for BodyContainsMatcher {
+    fn matches(&self, request: &Request) -> bool {
+        let body = str::from_utf8(&request.body)
+            .expect("can't convert body from byte slice to string")
+            .to_string();
+        let part =
+            str::from_utf8(&self.0).expect("can't convert expected part from byte slice to string");
```

Comment:
We don't want to have `expect`/`unwrap` here. Panicking in a matcher will tear down the entire MockActor.
If any of these operations fails, we should just return false.

Reply:
If you want to provide good diagnostics for debugging, a `log::debug / log::trace` statement would be ideal.